Usamos teste de hipotese para provar ou refutar algo

Uma máquina produz perfumes com média de 150 ml e desvio padrão de 2 ml. Quatro frascos foram recolhidos ao acaso da produção e o volume médio desses 4 frascos foi 153,8 ml. Essa média permite dizer que a máquina está produzindo dentro da especificação? Realize o teste com nível de confinça de 95%.

H0: μ = 150 ml \
Ha: μ ≠ 150ml


Alpha = 1 - 0,95 = 0,05  \
Z calc = (Média observada - média populacional) / (desvio padrão / raiz de n)

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import norm

In [3]:
z_calc = (153.8 - 150)/(2/np.sqrt(4))
z_calc

3.8000000000000114

### Funções da Distribuição Normal (Usando `scipy.stats.norm`)

- **`norm.pdf(x, loc=0, scale=1)`**: 
  - **Função densidade de probabilidade** (PDF).
  - Retorna a altura da função densidade no ponto `x`.
  - Para a normal padrão, `loc=0` e `scale=1` são os valores padrão (média e desvio padrão).

- **`norm.cdf(x, loc=0, scale=1)`**: 
  - **Função distribuição acumulada** (CDF).
  - Retorna a **probabilidade acumulada até o ponto `x`** (área sob a curva à esquerda de `x`).
  
- **`norm.sf(x, loc=0, scale=1)`**: 
  - **Função sobrevivência** (SF).
  - Retorna a probabilidade de o valor ser **maior que `x`**: \( 1 - CDF(x) \).
  
- **`norm.ppf(x, loc=0, scale=1)`**:
  - **Função ponto percentual** (Percent Point Function - PPF).
  - Retorna o valor de `x` para o qual a **probabilidade acumulada** é igual a `x`.
  - É a inversa da CDF, ou seja, o **percentil correspondente** a uma probabilidade dada.
  
- **`norm.isf(x, loc=0, scale=1)`**:
  - **Função inversa de sobrevivência** (Inverse Survival Function - ISF).
  - Retorna o valor de `x` tal que a probabilidade de ser **maior que `x`** é igual a `x`.
  - É a inversa da SF, ou seja, o valor para o qual a **probabilidade de sobrevivência** é dada.

In [4]:
norm.cdf(-1.96)

0.024997895148220435

In [6]:
norm.ppf(0.025)

-1.9599639845400545

In [7]:
norm.sf(1.96)

0.024997895148220435

In [9]:
norm.isf(0.025)

1.9599639845400545

Temos sz_criticos = 1.96 e z_calc = 3.8; portanto rejeitamos a hipotese nula, o volume medio é diferente de 150ml